snorql-extensions/src/main/kotlin/com/udaan/snorql/extensions/storage/metrics/DbIndexRedundancyMetric.kt

```diff
@@ -0,0 +1,273 @@
+                val similarIndexes = tableIndexes.filter {
+                    !indexesToSkip.contains(it) && isSimilar(indexInfo, it)
+                }
+                similarIndexes.forEach { similarIndex ->
+                    similarIndex.reason = RedundantReasonDTO(
+                        type = RedundantReasons.SIMILAR,
+                        servingIndex = indexInfo.indexName,
+                        message = "Index is SIMILAR to ${indexInfo.indexName}. Include missing Include columns in " +
+                                "${indexInfo.indexName}, post which this index can be deleted."
+                    )
+                    analyzedTableIndexes.add(similarIndex)
+                }
+                indexesToSkip.addAll(similarIndexes)
+                logger.debug("[analyseIndexes] Similar Indexes: $similarIndexes")
+                logger.debug("[analyseIndexes] Classified Indexes: $analyzedIndexes")
+            }
+            analyzedIndexes.addAll(analyzedTableIndexes)
+        }
+        return analyzedIndexes
+    }
+
+    private fun isDuplicate(parentIndex: DbIndexRedundancyDTO, childIndex: DbIndexRedundancyDTO): Boolean {
+        val parentIncludeCols = parentIndex.includeColumnNrs?.split(" ")?.toSet()
+        val childIncludeCols = childIndex.includeColumnNrs?.split(" ")?.toSet()
+        logger.debug(
+            "[DbIndexRedundancyMetric][isDuplicate] Parent Include Cols: $parentIncludeCols\n" +
+                    "Child Include Cols: $childIncludeCols"
+        )
+        if (parentIndex.indexColumnNrs == childIndex.indexColumnNrs && parentIncludeCols == childIncludeCols) {
+            return true
+        }
+        return false
+    }
+
+    private fun isOverlapping(parentIndex: DbIndexRedundancyDTO, childIndex: DbIndexRedundancyDTO): Boolean {
+        val parentIncludeCols =
+            parentIndex.includeColumnNrs?.split(" ")?.filterNot { it.isNullOrBlank() }?.toSet() ?: setOf()
+        val childIncludeCols =
+            childIndex.includeColumnNrs?.split(" ")?.filterNot { it.isNullOrBlank() }?.toSet() ?: setOf()
+        logger.debug(
+            "[DbIndexRedundancyMetric][isDuplicate] Parent Include Cols: $parentIncludeCols\n" +
+                    "Child Include Cols: $childIncludeCols"
+        )
+        if (checkStartsWithWord(
+                parentIndex.indexColumnNrs,
+                childIndex.indexColumnNrs
+            ) // && (parentIncludeCols subtract childIncludeCols).isEmpty()
+            && (childIncludeCols subtract parentIncludeCols).isEmpty()
+        ) {
+            return true
+        }
+        return false
+    }
+
+    private fun isSimilar(parentIndex: DbIndexRedundancyDTO, childIndex: DbIndexRedundancyDTO): Boolean {
+        if (parentIndex.indexColumnNrs == childIndex.indexColumnNrs
+            && parentIndex.includeColumnNrs != childIndex.includeColumnNrs
+        ) {
+            return true
+        }
+        return false
```

Comment:
`if (bool_condition) { return true } else { return false }` is a redundant expression.
Always use `return bool_condition`

Reply:
Resolved